storage/metrics_test.go

```diff
@@ -0,0 +1,183 @@
+package storage
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/kacperf531/sockchat/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testIndexName = "test_index"
+	channelA      = "foo"
+	channelB      = "bar"
+	authorNick    = "FooBar420"
+)
+
+var testChannels = []string{channelA, channelB}
+
+var (
+	timestampWithinPreviousMinute = time.Now().Unix() - 59
+	timestampWithinPreviousHour   = time.Now().Unix() - int64(60*time.Minute.Seconds())
+	timestampWithinPreviousDay    = time.Now().Unix() - int64(24*time.Hour.Seconds())
```

Comment:
We have races in tests because `time.Now()` is called in many places, a few are listed above and another one is called in `GetUserMessagesCounts` from `test_utils/test_doubles.go:128`
```go
MessageCountDistribution: []api.DistributionEntry{{PeriodStart: time.Now().Format("2006-01-02"), MessagesInPeriod: 69}},
```
Maybe these constants could be used in the stub? Currently we can get this from time to time:
```
--- FAIL: TestUserActivityReport (2.70s)
    --- FAIL: TestUserActivityReport/returns_results_grouped_by_minute (0.01s)
        metrics_test.go:144: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:144
                Error:          Not equal: 
                                expected: 1
                                actual  : 0
                Test:           TestUserActivityReport/returns_results_grouped_by_minute
        metrics_test.go:145: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:145
                Error:          Not equal: 
                                expected: "2023-05-24 10:16"
                                actual  : "2023-05-24 10:17"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -2023-05-24 10:16
                                +2023-05-24 10:17
                Test:           TestUserActivityReport/returns_results_grouped_by_minute
        metrics_test.go:145: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:145
                Error:          Not equal: 
                                expected: "2023-05-24 10:16"
                                actual  : "2023-05-24 10:17"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -2023-05-24 10:16
                                +2023-05-24 10:17
                Test:           TestUserActivityReport/returns_results_grouped_by_minute
```

Reply:
Also on the `:00` minutes of the hour I get:
```
--- FAIL: TestUserActivityReport (2.80s)
    --- FAIL: TestUserActivityReport/returns_results_grouped_by_hour (0.01s)
        metrics_test.go:124: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:124
                Error:          Not equal: 
                                expected: 1
                                actual  : 0
                Test:           TestUserActivityReport/returns_results_grouped_by_hour
        metrics_test.go:125: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:125
                Error:          Not equal: 
                                expected: "2023-05-24 10:00"
                                actual  : "2023-05-24 11:00"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -2023-05-24 10:00
                                +2023-05-24 11:00
                Test:           TestUserActivityReport/returns_results_grouped_by_hour
        metrics_test.go:127: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:127
                Error:          Not equal: 
                                expected: 0
                                actual  : 1
                Test:           TestUserActivityReport/returns_results_grouped_by_hour
        metrics_test.go:125: 
                Error Trace:    /Users/wojciechfabjanczuk/GolandProjects/sockchat/storage/metrics_test.go:125
                Error:          Not equal: 
                                expected: "2023-05-24 10:00"
                                actual  : "2023-05-24 11:00"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -2023-05-24 10:00
                                +2023-05-24 11:00
                Test:           TestUserActivityReport/returns_results_grouped_by_hour
```

Reply:
done in [apply review suggestions](https://github.com/kacperf531/sockchat/pull/11/commits/4054b76b6854361e07ae3021db901a9e9bd32cb8) - no longer coupling this report to current time/date